lib/rex/post/meterpreter/packet_response_waiter.rb

```diff
@@ -57,25 +64,14 @@ def notify(response)
   # If the interval is -1 we can wait forever.
   #
   def wait(interval)
-    if( interval and interval == -1 )
-      while(not self.done)
-        ::IO.select(nil, nil, nil, 0.1)
-      end
-    else
-      begin
-        Timeout.timeout(interval) {
-          while(not self.done)
-            ::IO.select(nil, nil, nil, 0.1)
-          end
-        }
-      rescue Timeout::Error
-        self.response = nil
-      end
-    end
+    interval = nil if interval and interval == -1
+    self.mutex.synchronize {
```

Comment:
I believe there is a race condition here. If a packet has arrived and triggered a `notify` before `wait` gets called, this will hang indefinitely waiting for a packet that has already come.

Also, as long as we're in here this should be using do/end per the style guide:
```ruby
  if response.nil?
    mutex.synchronize do
      cond.wait(mutex, interval)
    end
  end
  response